res/stasis/control.c

```diff
@@ -295,6 +308,12 @@ static struct stasis_app_command *exec_command_on_condition(
 	ast_cond_signal(&control->wait_cond);
 	ao2_unlock(control->command_queue);
 
+	if (control->control_thread != AST_PTHREADT_NULL) {
```

Comment:
@jcolp should the command_queue lock above be extended through this if?

Reply:
Yes